### PR TITLE
[5.1] Add validated() and unvalidated() method to Validator

### PR DESCRIPTION
**Pull request by manan-jadhav**
This PR is a followup on #8195.

The `validated()` method will return the data that : 
1. Had a validation rule
2. Passed the validation rule 

The `unvalidated()` method will return the data that : 
1. Did not have a validation rule

The `valid()` returns the data that : 
1. Passed the validation rule OR there was no validation rule for the datum

The `invalid()` returns the data that : 
1. Failed the validation rule.

The `valid()` and `invalid()` methods always existed, and this PR adds tests for these two methods.

### CODE REVIEW

File: src/Illuminate/Validation/Validator.php [hunk 1]
```diff
@@ -390,8 +390,29 @@ public function invalid()
...
+    
+    /**
+     * Returns the data which was validated, and passed.
+     *
+     * @return array
+     */
+    public function validated()
+    {
+        $valid = $this->valid();
+        return array_intersect_key($valid, $this->rules);
```
Comment (GrahamCampbell) [inline src/Illuminate/Validation/Validator.php]: Please just oneline this whole method.

File: tests/Validation/ValidationValidatorTest.php [hunk 2]
```diff
@@ -10,7 +10,35 @@ public function tearDown()
...
+        $this->assertEquals($v->invalid(),['bar' => 'notNumeric']);
+    }
 
+    public function testValidatedReturnsValidatedDataOnly()
+    {
+        $trans = $this->getRealTranslator();
+        $v = new Validator($trans, ['foo' => 'notEmpty','bar' => 'notNumeric','baz' => 'extra'], ['foo' => 'required|alpha', 'bar' => 'numeric']);
+        $this->assertEquals($v->validated(),['foo' => 'notEmpty']);
+    }
+
```
Comment (GrahamCampbell) [inline tests/Validation/ValidationValidatorTest.php]: no trailing whitespace please

File: tests/Validation/ValidationValidatorTest.php [hunk 3]
```diff
@@ -10,7 +10,35 @@ public function tearDown()
...
         m::close();
     }
+    
+    public function testValidReturnsValidData()
+    {
+        $trans = $this->getRealTranslator();
+        $v = new Validator($trans, ['foo' => 'notEmpty','bar' => 'notNumeric','baz' => 'extra'], ['foo' => 'required|alpha', 'bar' => 'numeric']);
+        $this->assertEquals($v->valid(),['foo' => 'notEmpty','baz' => 'extra']);
+    }
+
```
Comment (GrahamCampbell) [inline tests/Validation/ValidationValidatorTest.php]: same

File: tests/Validation/ValidationValidatorTest.php [hunk 4]
```diff
@@ -10,7 +10,35 @@ public function tearDown()
     {
         m::close();
     }
+
```
Comment (GrahamCampbell) [inline tests/Validation/ValidationValidatorTest.php]: same

**Comment by Sladewill** [comment]
This is exactly what the validator should do. @garygreen

**Comment by garygreen** [comment]
Does this work with arrays? Also still not convinced it needs to be two different methods. I think the original intend when `valid()` and `invalid()` was added was to provide data that was validated and ones that was invalid. This pr overcomplicates it when all that needs fixing is the `valid()` method...

**Comment by CurosMJ** [comment]
I would like to know why this was closed.

**Pull request closed without merging**